### PR TITLE
added authentication, added proxy support

### PR DESCRIPTION
**Pull request by acole76**
Added the ability to login to expireddomains.net.  Added the ability to pass in proxy as a switch.
> python.exe .\domainhunter.py -u USERNAME -p PASSWORD -P https://127.0.0.1:8080

New Switches are:
-u for the username
-p for the password
-P (caps) for the proxy

if you don't specify -p for the password, you will be prompted for it.  There seems to be an issue with expireddomains.net, they do not appear to ever return "expired" domains, so the keyword searches are always empty.  idk what the original intent was for that feature, so I left it as it was.

### CODE REVIEW

**Comment by acole76** [comment]
I got sloppy w/ my commits.  Items I added:

--keyword-start search domains that start with a string
--keyword-end search domain that end with a string
--output save to a specific file

keyword search returns only available domains.  searches can return up to 200 rows, optimized code to break out of loop if there are no more records to fetch.

File: domainhunter.py [hunk 1]
```diff
@@ -609,12 +622,17 @@ def getIndex(cells, index):
...
+		unwantedResults = ['Uncategorized','error','Not found.','Spam','Spam URLs','Pornography','badurl','Suspicious','Malicious Sources/Malnets','captcha','Phishing','Placeholders']
+		
                 bluecoat = checkBluecoat(domain)
-                print("[+] {}: {}".format(domain, bluecoat))
+		if bluecoat not in unwantedResults:
+                    print("[+] Bluecoat - {}: {}".format(domain, bluecoat))
                 ibmxforce = checkIBMXForce(domain)
-                print("[+] {}: {}".format(domain, ibmxforce))
+		if ibmxforce not in unwantedResults:
+                    print("[+] IBM XForce - {}: {}".format(domain, ibmxforce))
```
Comment (andrewchiles) [inline domainhunter.py]: Not Python3 syntax (change to use spaces vs tabs)
```./domainhunter.py -s expireddomains.net
  File "./domainhunter.py", line 628
    if bluecoat not in unwantedResults:```

File: domainhunter.py [hunk 2]
```diff
@@ -609,12 +622,17 @@ def getIndex(cells, index):
             # Perform domain reputation checks
             if check:
                 
+		unwantedResults = ['Uncategorized','error','Not found.','Spam','Spam URLs','Pornography','badurl','Suspicious','Malicious Sources/Malnets','captcha','Phishing','Placeholders']
+		
                 bluecoat = checkBluecoat(domain)
-                print("[+] {}: {}".format(domain, bluecoat))
+		if bluecoat not in unwantedResults:
+                    print("[+] Bluecoat - {}: {}".format(domain, bluecoat))
```
Comment (andrewchiles) [inline domainhunter.py]: Not Python3 syntax (change to use spaces vs tabs)
```./domainhunter.py -s expireddomains.net
  File "./domainhunter.py", line 628
    if bluecoat not in unwantedResults:```

File: domainhunter.py [hunk 3]
```diff
@@ -595,6 +607,7 @@ def getIndex(cells, index):
         # Print number of domains to perform reputation checks against
         if check:
             print("\n[*] Performing reputation checks for {} domains".format(len(domain_list_unique)))
+	    print("")
```
Comment (andrewchiles) [inline domainhunter.py]: Not Python3 syntax (change to use spaces vs tabs)
```python3 ./domainhunter.py
  File "./domainhunter.py", line 610
    print("")```

File: domainhunter.py [hunk 4]
```diff
@@ -18,6 +18,7 @@
 import sys
 import urlparse
 import getpass
+import uuid
```
Comment (andrewchiles) [inline domainhunter.py]: requirements.txt must be updated to include new packages `urlparse` `getpass` `uuid`

File: domainhunter.py [hunk 5]
```diff
@@ -18,6 +18,7 @@
 import sys
 import urlparse
```
Comment (andrewchiles) [inline domainhunter.py]: Modify line to correct import the Python3 library `from urllib.parse import urlparse`

File: domainhunter.py (hunk 1 above)
Comment (acole76) [reply domainhunter.py]: tabs fixed

File: domainhunter.py (hunk 2 above)
Comment (acole76) [reply domainhunter.py]: tabs fixed

File: domainhunter.py (hunk 3 above)
Comment (acole76) [reply domainhunter.py]: tabs fixed

File: domainhunter.py (hunk 5 above)
Comment (acole76) [reply domainhunter.py]: syntax updated

Comment (acole76) [reply domainhunter.py]: reqs updated, getpass is part of the python core.

**Comment by acole76** [comment]
1. keyword gets passed in via the -k arg and is represented in the URL as "fdomain", -ks and -ke are completely optional and shouldn't affect the search results if they are left blank
2. The changes I made to bluecoat were related to the XSRF.  I did not make any changes to the OCR, though, I have never personally seen OCR work.
3. I'm not sure I follow this, I see categorization when I run it

image

**Comment by andrewchiles** [comment]
Thanks for the quick response!

For #3 ( -s option), categorization is displayed but you lose visibility into the reputation source. If your target org only uses Bluecoat, for example, then that’s all you really care about.  The output doesn’t make it apparent where the reputation is coming from. Should be easy enough to add into the wrapper function for that command.

**Comment by acole76** [comment]
I see, the print statements were commented out.  I put them all back in.

image

**Pull request merged**